src/metatensor/models/utils/normalize.py

```diff
@@ -0,0 +1,89 @@
+from typing import List, Union
+
+import torch
+from metatensor.learn.data.dataset import _BaseDataset
+from metatensor.torch import TensorBlock, TensorMap
+
+
+def get_average_number_of_atoms(
+    datasets: List[Union[_BaseDataset, torch.utils.data.Subset]]
+):
+    """Calculate the average number of atoms in a dataset."""
+    average_number_of_atoms = []
+    for dataset in datasets:
+        num_atoms = []
+        for i in range(len(dataset)):
+            structure = dataset[i].structure
+            if not isinstance(structure, torch.ScriptObject):
+                raise RuntimeError
+            num_atoms.append(len(structure))
+        average_number_of_atoms.append(
+            torch.mean(torch.tensor(num_atoms).to(torch.get_default_dtype()))
+        )
+    return torch.tensor(average_number_of_atoms)
+
+
+def get_average_number_of_neighbors(
+    datasets: List[Union[_BaseDataset, torch.utils.data.Subset]]
+):
+    """Calculate the average number of neighbors in a dataset."""
+    average_number_of_neighbors = []
+    for dataset in datasets:
+        num_neighbors = []
+        for i in range(len(dataset)):
+            structure = dataset[i].structure
+            if not isinstance(structure, torch.ScriptObject):
+                raise RuntimeError
```

Comment:
We need a small description of the error

Reply:
Same